test/topology_experimental_raft/test_raft_voters.py

```
@@ -0,0 +1,90 @@
+    """
+
+    config = {
+        'endpoint_snitch': 'GossipingPropertyFileSnitch',
+    }
+    dc_setup = [
+        {
+            'property_file': {'dc': 'dc1', 'rack': 'rack'},
+            # The large DC has 2x the number of nodes of the smaller DCs
+            'num_nodes': 2 * num_nodes,
+        },
+        {
+            'property_file': {'dc': 'dc2', 'rack': 'rack'},
+            'num_nodes': num_nodes,
+        },
+        {
+            'property_file': {'dc': 'dc3', 'rack': 'rack'},
+            'num_nodes': num_nodes,
+        },
+    ]
+
+    # Arrange: create DCs / servers
+
+    dc_servers = []
+    for dc in dc_setup:
+        logging.info(f"Creating {dc['property_file']['dc']} with {dc['num_nodes']} nodes")
+        dc_servers.append(await manager.servers_add(dc['num_nodes'], config=config,
+                                                    property_file=dc['property_file']))
+
+    assert len(dc_servers) == len(dc_setup)
+
+    logging.info('Creating connections to all DCs')
+    dc_cqls = []
+    for servers in dc_servers:
+        dc_cqls.append(cluster_con([servers[0].ip_addr], 9042, False,
+                                   load_balancing_policy=WhiteListRoundRobinPolicy([servers[0].ip_addr])).connect())
+
+    assert len(dc_cqls) == len(dc_servers)
+
+    # Act: Kill all nodes in dc1
+
+    # Note: This also tests the topology coordinator migration
+    # (the coordinator will be most likely in dc1 initially, but after the nodes are killed, it should migrate to one of
+    # the other DCs - testing that the voters status is transferred correcly)
+
+    logging.info('Killing down all nodes in dc1')
+    await asyncio.gather(*(manager.server_stop_gracefully(srv.server_id) for srv in dc_servers[0]))
+
+    # Assert: Verify that the majority has not been lost (we can change the topology)
+
+    logging.info('Executing read barrier')
+    await read_barrier(manager.api, dc_servers[1][0].ip_addr)
```

For how long the test will hang before reporting an error in case of the failure?

---

It hangs for 1 minute (the raft timeout) ... could maybe set a shorter timeout to make it fail faster (and then remove with the feature ready)

---

1 minute is fine. I was afraid it will hang for test.py timeout which is much longer.